In [1]:
import numpy as np
import pandas as pd

In [2]:
# The statistical module used to run chi square test
import scipy.stats as stats

In [3]:
# Observed data in a (hypothetical) survey of 6000 people 
observed = pd.Series([1500, 1350, 1600, 1550], index=["1", "2", "3", "4"])


In [10]:
# Create a data frame
df = pd.DataFrame([observed]).T

In [23]:
# Add a column whose default values are the expected values
df[1] = 1500

In [25]:
# Rename the data frame columns
df.columns = ['Daily Visits','Expected']
df

Daily Visits  Expected
1          1500      1500
2          1350      1500
3          1600      1500
4          1550      1500

In [ ]:
# View the data frame


In [34]:
# Calculate the critical value
critical_value = stats.chi2.ppf(q = 0.95, df = 3)

In [35]:
# Print the critical value
critical_value

7.814727903251179

In [36]:
# Run the chi square test with stats.chisquare()
stats.chisquare(df['Daily Visits'], df['Expected']) 

Power_divergenceResult(statistic=23.333333333333336, pvalue=3.4411276663397364e-05)

 Conclusion

Since the chi square value of 23.3 exceeds the critical value of 7.81, we conclude that the results are statistically significant.